core/application/impl/app_configuration_impl.cpp

```diff
@@ -98,6 +101,22 @@ namespace {
     }
     return std::nullopt;
   }
+
+  std::optional<kagome::primitives::BlockId> str_to_recovery_state(
+      std::string_view str) {
+    kagome::primitives::BlockNumber bn;
+    auto result = std::from_chars(str.data(), str.data() + str.size(), bn);
```

Comment:
It is not really a reliable way to distinguish between a number and a hex-string.

Here would be the case with the hex-string containing only digits.

The possible fix could be to restrict hex-string format to only those which start with 0x prefix.

Reply:
In CLI hex is used  without heading 0x. But I agree, it is reliable to try read it as block hash first